ciao-controller/compute.go

```diff
@@ -461,24 +478,24 @@ func showServerDetails(w http.ResponseWriter, r *http.Request, context *controll
 	dumpRequest(r)
 
 	if validateToken(context, r) == false {
-		http.Error(w, "Invalid token", http.StatusInternalServerError)
+		returnErrorCode(w, http.StatusUnauthorized, "Invalid token")
 		return
 	}
 
 	instance, err := context.ds.GetInstance(instanceID)
 	if err != nil {
-		http.Error(w, "Instance not available", http.StatusInternalServerError)
+		returnErrorCode(w, http.StatusNotFound, "Instance could not be found")
 		return
 	}
 
 	if instance.TenantID != tenant {
-		http.Error(w, "Instance not available", http.StatusInternalServerError)
+		returnErrorCode(w, http.StatusNotFound, "Instance not in tenant")
```

Comment:
I would replace "Instance not in tenant" with "Instance does not belong to tenant".